src/openassetio-python/CMakeLists.txt

```diff
@@ -112,15 +113,19 @@ if (OPENASSETIO_ENABLE_TESTS AND Python_Interpreter_FOUND)
         OUTPUT
         "${venv_python_dir}/pytest"
         COMMAND
-        "${installed_pip}" install -r "${PROJECT_SOURCE_DIR}/tests/requirements.txt"
+        "${installed_pip}" download --destination-directory ${download_dir} -r "${PROJECT_SOURCE_DIR}/tests/requirements.txt"
+        COMMAND
+        "${installed_pip}" wheel --wheel-dir ${download_dir} setuptools wheel
+        COMMAND
+        "${installed_pip}" install  --no-index --find-links ${download_dir} -r "${PROJECT_SOURCE_DIR}/tests/requirements.txt"
         DEPENDS "${installed_pip}"
     )
 
     # Target to install the pure Python component of the project.
     add_custom_target(
         openassetio-python-py-install
-        COMMAND "${installed_pip}" install "${PROJECT_SOURCE_DIR}"
-        DEPENDS "${installed_pip}"
+        COMMAND "${installed_pip}" install  --no-index --find-links ${download_dir} "${PROJECT_SOURCE_DIR}"
+        DEPENDS "${installed_pip}" "${venv_python_dir}/pytest"
```

Comment:
[question] Is this additional dependency required?

Reply:
It was required, the build can fail because the install requires those downloaded files and without the dependency this step may run before it, and failing obviously (I tried this out).

Reply:
It would be cleaner to put the package download into a separate step, which could also depend on the requirements.txt, and then let both the pytest- and openassetio-installer step depend on that one. Not hard to do and more maintainable. I will change it.

Reply:
Thanks - sounds like a good plan - then it's really clear about the coupling between the stages.

Reply:
Addressed in https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/309/commits/d5e0db78ef4e78635c7d3343eb825a936d2ccaa6